### ISSUE TITLE
Incorrect inspection "PHPUnit: bugs and best practices"

### ISSUE THREAD

**Issue by neuberkauf** [kalessil/phpinspectionsea#1830]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.7.1 |
| Language level | PHP 7.2 |
| Inspection | "PHPUnit: bugs and best practices" |
| Options | ![image](https://user-images.githubusercontent.com/103447747/171545751-dd108145-3345-4109-b342-9b49093bbb47.png) <br>Selecting 8.5 won't stick |

Current behaviour
--

When making a test dependent on a test in another class, inspections seem to get confused. The correct way to define the dependency is flagged as invalid.

Expected behaviour
--
The functional version of the `@depends` shouldn't flag the test method as using an invalid reference.

Environment
---

PHPUnit v. 8.5.26

What steps will reproduce the problem?
----

1. Create two test classes (in my case they're in the same namespace but separate class files):
   ```php
   namespace Test\Api\Uploads {
   class ATest extends PHPUnit\Framework\TestCase {
       public function testFoo () : string { return 'FOO'; }
   }
   class BTest extends PHPUnit\Framework\TestCase {
       public function testBar () : void { }
   }
   }
   ```
2. Make `BTest::testBar()` dependent on `ATest::testFoo()`. i've tried different variations of this, but only one way works...
   1. Same namespace, skip the FQN: this does not work, but inspections are clean.
   ```
   /**
    * @depends ATest::testFoo
    */
   public function testBar (string $response) : void { }
   ```
   2. Full FQN with root: this does not work, but inspections are clean.
   ```
   /**
    * @depends \Test\Api\Uploads\ATest::testFoo
    */
   public function testBar (string $response) : void { }
   ```
   3.  Full FQN without root: this ***DOES *** work, but inspections complain about two things:
   ```
   /**
    * @depends Test\Api\Uploads\ATest::testFoo
    */
   public function testBar (string $response) : void { }
   ```
      1. The namespace is highlighted at `Uploads` saying it doesn't exist.
      2. The test method `testBar` is highlighted as invalid
         ("@depends referencing to a non-existing or inappropriate entity")

What is the expected result?
----

The version that works should not complain about either of the two issues

What happens instead?
----

Misleading inspection results

Please provide any additional information below.
----

1. Screenshot to illustrate the inspection issues, yet XDebug was able to enter the function when i ran the test group
2. Logs & diagnostics zip attached

Note: The namespace inspection belongs to PhpStorm, which i reported to them.
https://youtrack.jetbrains.com/issue/WI-66961
i wanted to keep the issues together since they seem related in some way.

![false-positive-inspections](https://user-images.githubusercontent.com/103447747/171544497-ce65cf7d-6280-434f-82c4-f7ccb0740913.png)

Information from Help - About dialog
---

```
PhpStorm 2022.2 EAP
Build #PS-222.2680.8, built on May 27, 2022
Licensed to PhpStorm EAP user: Joe T
Expiration date: June 26, 2022
Runtime version: 17.0.3+7-b463.3 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Registry:
    debugger.watches.in.variables=false
    run.processes.with.pty=TRUE

Non-Bundled Plugins:
    br.com.matheusfm.jsonutils (1.2)
    one.util.ideaplugin.screenshoter (1.7.1)
    com.vecheslav.darculaDarkerTheme (1.2.0)
    com.github.leomillon.uuidgenerator (4.5.1)
    net.seesharpsoft.intellij.plugins.csv (2.18.2)
    com.markskelton.one-dark-theme (5.6.0)
    dev.patrickpichler.darculaPitchBlackTheme (1.0.0)
    de.netnexus.camelcaseplugin (3.0.12)
    krasa.CpuUsageIndicator (1.14)
    com.wix.scss.lint (2.1.1)
    com.intellij.ideolog (203.0.30.0)
    atom-one (1.5.1)
    com.vecheslav.darculaSolidTheme (1.0.3)
    MHCommitMessageTemplate (1.0.3)
    com.arcticicestudio.nord.jetbrains (0.13.0)
    mobi.hsz.idea.gitignore (4.4.1)
    com.kalessil.phpStorm.phpInspectionsEA (4.0.7.1)
    ru.adelf.idea.dotenv (2022.2)
    com.aurimasniekis.phppsr4namespacedetector (1.0.1)
    de.espend.idea.laravel (0.15.4)
```
